include/maidsafe/routing/node_info.h

```diff
@@ -34,43 +34,36 @@ namespace routing {
 struct NodeInfo {
   using PublicPmid = passport::PublicPmid;
 
-  NodeInfo(NodeId id, PublicPmid dht_fob) : id(id), dht_fob(std::move(dht_fob)) {}
-  NodeInfo() = default;
+  NodeInfo(NodeId id_in, PublicPmid dht_fob_in, bool connected_in)
+      : id(id_in), dht_fob(std::move(dht_fob_in)), connected(connected_in) {}
+
+  NodeInfo() : id(), dht_fob(), connected(false) {}
+
   NodeInfo(const NodeInfo&) = default;
 
   NodeInfo(NodeInfo&& other) MAIDSAFE_NOEXCEPT : id(std::move(other.id)),
-                                                 dht_fob(std::move(other.dht_fob)) {}
+                                                 dht_fob(std::move(other.dht_fob)),
+                                                 connected(std::move(other.connected)) {}
+
   NodeInfo& operator=(const NodeInfo&) = default;
 
   NodeInfo& operator=(NodeInfo&& other) MAIDSAFE_NOEXCEPT {
     id = std::move(other.id);
     dht_fob = std::move(other.dht_fob);
+    connected = std::move(other.connected);
     return *this;
   }
 
   bool operator==(const NodeInfo& other) const { return id == other.id; }
-  bool operator!=(const NodeInfo& other) const { return !operator==(other); }
+  bool operator!=(const NodeInfo& other) const { return id != other.id; }
```

Comment:
It was to keep all comparison operators consistent.  We only need to define `==` and `<` to be able to define all the others in terms of these, but we weren't doing that.  We had some that did and some that didn't, so I changed to all just use `id`.